### PR TITLE
Double slash fix

### PR DESCRIPTION
**Pull request by sblaisot**
This addresses lp:1464975
Replace all (at least most of) occurences of a manual path construction by string concatenation by the use of QDir::filePath.

This way, we avoid double-slahs and can handle properly settingsPath both with or without a trailing "/".

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -95,14 +95,14 @@ void MessageHandler(QtMsgType type,
...
+                logFileName = QDir(logLocation).filePath("mixxx.log");
             } else {
-                logFileName = QString("%1/mixxx.log.%2").arg(logLocation).arg(i);
+                logFileName = QDir(logLocation).filePath(QString("mixxx.log.%1").arg(i));
             }
             QFileInfo logbackup(logFileName);
             if (logbackup.exists()) {
                 QString olderlogname =
-                        QString("%1/mixxx.log.%2").arg(logLocation).arg(i + 1);
+                        QDir(logLocation).filePath(QString("mixxx.log.%1").arg(i + 1));
```
Comment (daschuer) [inline src/main.cpp]: use append() or just +

Comment (daschuer) [inline src/main.cpp]: Ah nevermind I was wrong, It will be ::number() but that's more code to read.

File: src/util/cmdlineargs.h [hunk 2]
```diff
@@ -64,7 +64,9 @@ class CmdlineArgs {
     bool getTimelineEnabled() const { return !m_timelinePath.isEmpty(); }
     const QString& getLocale() const { return m_locale; }
     const QString& getSettingsPath() const { return m_settingsPath; }
-    void setSettingsPath(QString newSettingsPath) { m_settingsPath=QString(newSettingsPath); return; }
+    void setSettingsPath(const QString& newSettingsPath) { 
+        m_settingsPath=newSettingsPath;
```
Comment (daschuer) [inline src/util/cmdlineargs.h]: nit: m_settingsPath = newSettingsPath;

**Comment by daschuer** [comment]
Can we remove the trailing "/" form getSettingsPath() now?

**Comment by daschuer** [comment]
This is a lot to test. :-/ is it save for 1.12?

File: src/util/cmdlineargs.h (hunk 2 above)
Comment (sblaisot) [inline src/util/cmdlineargs.h]: ok, updated.

**Comment by sblaisot** [comment]
Do you mean this PR should be against master ?
The tests will mostly be needed if we remove trailing /. I kept it for now, we can remove it in later version.

**Comment by daschuer** [comment]
OK, the log file issues seams to be fixed now.
Can you add a comment that the tailing / is a candidate to remove? 
@all: merge to 1.12?

**Comment by sblaisot** [comment]
ok, comments added.

**Comment by daschuer** [comment]
LGTM Thank you very much.

**Pull request merged**